Sourcery/Models/Type.swift

```diff
@@ -14,10 +14,15 @@ class Type: NSObject {
 
     /// Name in global scope 
     var name: String {
-        guard let parentName = parentName else { return localName }
+        guard let parentName = parent?.name else { return localName }
         return "\(parentName).\(localName)"
     }
 
+    /// Is this type generic? regardless whether it's this type directly or one of its parent types
+    var isGeneric: Bool {
+        return (parent?.isGeneric ?? false) || hasGenericComponent
```

Comment:
Do you think we should remove this check to improve performance, or doesn't matter?